libcrux-ml-dsa/src/simd/portable/encoding/commitment.rs

```diff
@@ -127,9 +135,17 @@ pub fn serialize_6(simd_unit: &Coefficients, serialized: &mut [u8]) {
                 == {} get_bit (Seq.index $coefficients (n / 6)) (sz (n % 6));
                 == {} bit_vec_of_int_t_array #I32 #(mk_usize 4) $coefficients 6 n;
                 == {} out (v i * 24 + n);
-                })
+                });
+           (* Find a better style for these sequence updates so that such silly assertions become unnecessary *)
+           assert (forall (n:nat{n >= 24 * v i /\ n < 24 * v i + 24}). inp (24 * v i + (n - 24 * v i)) == out (24 * v i + (n - 24 * v i)));
+           assert (forall (n:nat{n >= 24 * v i /\ n < 24 * v i + 24}). inp n == out n);
+           assert (forall (n:nat{n < v i * 24}). n / 8 < 3 * v i);
+           assert (forall (j:nat{j < 3 * v i}). Seq.index ${serialized} j == Seq.index (Seq.slice ${serialized} 0 (3 * v i)) j);
+           assert (forall (j:nat{j < 3 * v i}). Seq.index ${_old_serialized} j == Seq.index (Seq.slice ${_old_serialized} 0 (3 * v i)) j);
+           assert (forall (n:nat{n < 24 * (v i + 1)}). inp n == out n))
```

Comment:
We need both the calc proof and the assertions? I guess that's because of the `serialized[*..*]`?

Reply:
I wonder if we should just not have a lemma that generalize "updating contiguous blocks of an array in a loop".
I will take a look at the other portable serialize. That pattern isprobably everywhere, so it might be worth writing such a lemma

Reply:
Let's add any improvements as notes to https://github.com/cryspen/libcrux/issues/777